src/engine.cairo

```diff
@@ -110,11 +120,16 @@ pub impl EngineImpl of EngineTrait {
         Opcode::is_opcode_always_illegal(opcode, ref self)?;
 
         if !self.cond_stack.branch_executing() && !flow::is_branching_opcode(opcode) {
+            // check stack size
+            self.check_stack_size()?;
             Opcode::is_opcode_disabled(opcode, ref self)?;
             self.opcode_idx += 1;
             return Result::Ok(true);
         }
 
+        // check stack size
+        self.check_stack_size()?;
```

Comment:
It might be better to check the stack size after executing, so moving it to like 135

Reply:
any reason for this, i though, it will be better we check the stack size before executing the opcode